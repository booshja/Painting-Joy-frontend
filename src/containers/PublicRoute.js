// dependencies
import React, { useContext } from "react";
import styled from "styled-components";
// components
import { Route } from "react-router-dom";
import { Header } from "./";
import { Footer, TitleLogo } from "../components";
// context
import MenuContext from "../context/MenuContext";
// breakpoints
import { breakpoints } from "../breakpoints";

const StyledBackground = styled.div`
    width: 100%;
    min-height: 100vh;

    ${breakpoints("background-color", "", [{ 1024: "#ffffff" }])}
    ${breakpoints("width", "px", [{ 1024: 770 }])}
    ${breakpoints("max-width", "px", [{ 1024: 780 }])}
`;

const StyledMain = styled.main``;

const PublicRoute = ({ children, ...rest }) => {
    // set up context
    const { menuOpen } = useContext(MenuContext);

    return menuOpen ? (
        <Header />
    ) : (
        <StyledBackground>
            <Header />
            <StyledMain>
                <TitleLogo />
                <Route {...rest} render={() => children} />
            </StyledMain>
            <Footer />
        </StyledBackground>
    );
};

export default PublicRoute;
