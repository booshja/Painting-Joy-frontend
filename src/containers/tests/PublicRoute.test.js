import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import PublicRoute from "../PublicRoute";
import MenuContext from "../../context/MenuContext";
import CartContext from "../../context/CartContext";
import "jest-styled-components";

/** Smoke Test */
it("renders without crashing", () => {
    render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: false }}>
                <CartContext.Provider value={{ cart: [1, 2, 3] }}>
                    <PublicRoute />
                </CartContext.Provider>
            </MenuContext.Provider>
        </BrowserRouter>
    );
});

/** Snapshot Test */
it("matches snapshot", () => {
    const { asFragment } = render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: false }}>
                <CartContext.Provider value={{ cart: [1, 2, 3] }}>
                    <PublicRoute />
                </CartContext.Provider>
            </MenuContext.Provider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

/** RTL Tests */
it("renders correctly", () => {
    const { getByText } = render(
        <BrowserRouter>
            <MenuContext.Provider value={{ menuOpen: false }}>
                <CartContext.Provider value={{ cart: [1, 2, 3] }}>
                    <PublicRoute children={<h1>Children!</h1>} />
                </CartContext.Provider>
            </MenuContext.Provider>
        </BrowserRouter>
    );

    expect(getByText("Children!")).toBeInTheDocument();
});
