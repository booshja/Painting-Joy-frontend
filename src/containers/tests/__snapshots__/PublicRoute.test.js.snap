// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 2rem;
  height: 2rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 101;
}

.c3:focus {
  outline: none;
}

.c3 div {
  width: 2rem;
  height: 0.3rem;
  background-color: #207070;
  border-radius: 10px;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
  position: relative;
  -webkit-transform-origin: 1px;
  -ms-transform-origin: 1px;
  transform-origin: 1px;
}

.c3 div:first-child {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c3 div:nth-child(2) {
  opacity: 1;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
}

.c3 div:nth-child(3) {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c16 {
  width: 150px;
  height: 150px;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 125px;
}

.c18 {
  font-family: "News Cycle",sans-serif;
  font-size: 1.2rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  color: #ffffff;
}

.c11 {
  background-color: #012e34;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  color: #ffffff;
  padding: 1rem;
  min-height: 150px;
  margin-top: 2rem;
}

.c12 {
  font-family: "Barlow Condensed",sans-serif;
  font-weight: 700;
  font-size: 1.4rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c13 {
  font-family: "News Cycle",sans-serif;
  font-size: 1.1rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c19 {
  font-family: "News Cycle",sans-serif;
  font-size: 0.6rem;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
}

.c14 {
  margin: 1rem 0 2rem 0;
}

.c4 {
  background-image: url(wall_image_complete.jpg);
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
  z-index: 100;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  width: 100%;
  text-align: center;
  position: absolute;
  top: 0;
  left: 0;
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
  -webkit-transform: translateY(-100%);
  -ms-transform: translateY(-100%);
  transform: translateY(-100%);
}

.c6 {
  font-family: "News Cycle",sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  font-size: 2rem;
  text-transform: uppercase;
  padding: 1rem 0;
  color: #ffffff;
  -webkit-transition: color 0.3s linear;
  transition: color 0.3s linear;
}

.c6.active {
  font-weight: 700;
}

.c5 {
  background-color: rgba(0,0,0,0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1rem;
}

.c10 {
  font-family: "News Cycle",sans-serif;
  font-weight: 400;
  font-size: 1.4rem;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
}

.c10:hover:not(.active) {
  -webkit-transform: translateY(-1px);
  -ms-transform: translateY(-1px);
  transform: translateY(-1px);
}

.c10.active {
  font-weight: 700;
  color: #207070;
}

.c9 {
  display: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 1rem;
  width: 55%;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 1rem;
}

.c8 {
  width: 250px;
  height: 250px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 60px;
  padding: 0 1rem;
  border-bottom: 1px solid #207070;
}

.c2 {
  font-family: "Barlow Condensed",sans-serif;
  font-size: 1.8rem;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  color: #207070;
}

.c0 {
  width: 100%;
  min-height: 100vh;
}

@media screen and (min-width:414px) {
  .c12 {
    font-size: 1.8rem;
  }
}

@media screen and (min-width:1024px) {
  .c12 {
    display: none;
  }
}

@media screen and (min-width:414px) {
  .c13 {
    font-size: 1.4rem;
  }
}

@media screen and (min-width:1024px) {
  .c13 {
    display: none;
  }
}

@media screen and (min-width:414px) {
  .c19 {
    font-size: 0.8rem;
  }
}

@media screen and (min-width:360px) {
  .c14 {
    display: none;
  }
}

@media screen and (min-width:1024px) {
  .c14 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (min-width:1024px) {
  .c14 {
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media screen and (min-width:1024px) {
  .c14 {
    width: 60%;
  }
}

@media screen and (min-width:414px) {
  .c6 {
    font-size: 2.4rem;
  }
}

@media screen and (min-width:768px) {
  .c6 {
    font-size: 3rem;
  }
}

@media screen and (min-width:1024px) {
  .c9 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (min-width:1024px) {
  .c7 {
    margin-top: 0;
  }
}

@media screen and (min-width:414px) {
  .c8 {
    width: 300px;
  }
}

@media screen and (min-width:768px) {
  .c8 {
    width: 350px;
  }
}

@media screen and (min-width:414px) {
  .c8 {
    height: 300px;
  }
}

@media screen and (min-width:768px) {
  .c8 {
    height: 350px;
  }
}

@media screen and (min-width:1024px) {
  .c1 {
    display: none;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    background-color: #ffffff;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    width: 770px;
  }
}

@media screen and (min-width:1024px) {
  .c0 {
    max-width: 780px;
  }
}

<div
    class="c0"
  >
    <header
      class="c1"
    >
      <a
        class="c2"
        href="/"
      >
        Painting Joy Mural Co.
      </a>
      <button
        class="c3"
      >
        <div
          data-testid="div1"
        />
        <div
          data-testid="div2"
        />
        <div
          data-testid="div3"
        />
      </button>
      <nav
        class="c4"
      >
        <div
          class="c5"
        >
          <a
            aria-current="page"
            class="c6 active"
            href="/"
          >
            Home
          </a>
          <a
            class="c6"
            href="/murals"
          >
            Murals
          </a>
          <a
            class="c6"
            href="/contact"
          >
            Contact Me
          </a>
          <a
            class="c6"
            href="/store"
          >
            Store
          </a>
          <a
            class="c6"
            href="/cart"
          >
            Cart (3)
          </a>
        </div>
      </nav>
    </header>
    <main
      class=""
    >
      <div
        class="c7"
      >
        <a
          href="/"
        >
          <img
            alt="Painting Joy Mural Co. Logo"
            class="c8"
            src="painting-joy-logo-transparent.png"
          />
        </a>
        <nav
          class="c9"
        >
          <a
            aria-current="page"
            class="c10 active"
            href="/"
          >
            Home
          </a>
          <a
            class="c10"
            href="/murals"
          >
            Murals
          </a>
          <a
            class="c10"
            href="/contact"
          >
            Contact
          </a>
          <a
            class="c10"
            href="/store"
          >
            Store
          </a>
          <a
            class="c10"
            href="/cart"
          >
            Cart (3)
          </a>
        </nav>
      </div>
    </main>
    <footer
      class="c11"
    >
      <a
        class="c12"
        href="/"
      >
        Painting Joy Mural Co.
      </a>
      <a
        class="c13"
        href="/contact"
      >
        Contact Me
      </a>
      <div
        class="c14"
      >
        <div
          class="c15"
        >
          <a
            href="/"
          >
            <img
              alt="Painting Joy Mural Co. Logo"
              class="c16"
              src="painting-joy-logo-white.jpg"
            />
          </a>
        </div>
        <div
          class="c17"
        >
          <a
            class="c18"
            href="/"
          >
            Home
          </a>
          <a
            class="c18"
            href="/murals"
          >
            Murals
          </a>
          <a
            class="c18"
            href="/contact"
          >
            Contact
          </a>
          <a
            class="c18"
            href="/store"
          >
            Store
          </a>
        </div>
      </div>
      <p
        class="c19"
      >
        Copyright © 2021 - Painting Joy Mural Co.
      </p>
    </footer>
  </div>
</DocumentFragment>
`;
